### PR TITLE
fix golem spawning on top of player

### PR DESCRIPTION
**Pull request by qndel**
Resolves https://github.com/diasurgical/devilutionX/issues/2833

### CODE REVIEW

**Review by StephenCWills — APPROVED** [review]

**Comment by AJenbo** [comment]
Could also use the MonsterPosOk function to allow them on non solid objects and doors :shrug:

File: Source/missiles.cpp [hunk 1]
```diff
@@ -2377,7 +2377,7 @@ void AddGolem(Missile &missile, Point dst, Direction /*midir*/)
 				if (!LineClearMissile(missile.position.start, target))
 					continue;
 
-				if (dMonster[target.x][target.y] != 0 || nSolidTable[dPiece[target.x][target.y]] || dObject[target.x][target.y] != 0)
+				if (dMonster[target.x][target.y] != 0 || nSolidTable[dPiece[target.x][target.y]] || dObject[target.x][target.y] != 0 || dPlayer[target.x][target.y] != 0)
```
Comment (julealgon) [inline Source/missiles.cpp:2380]: Don't we have a "tile is free" function somewhere we could reuse here?

Comment (StephenCWills) [reply Source/missiles.cpp:2380]: Maybe [CanPlaceMonster()](https://github.com/diasurgical/devilutionX/blob/0d3bf15601ab2a45464c72ec115673f7833acdc5/Source/monster.cpp#L265) or [IsTileAvailable()](https://github.com/diasurgical/devilutionX/blob/0d3bf15601ab2a45464c72ec115673f7833acdc5/Source/monster.cpp#L2015). I don't see a `MonsterPosOk()` function anywhere.

* The former checks dungeon bounds, `dMonster`, `dPlayer`, `dFlags` for `BFLAG_VISIBLE` and `BFLAG_POPULATED`, and `nSolidTable`, but not objects.
* The latter checks `dMonster`, `dPlayer`, `dObject`, and `nSolidTable`, but doesn't check dungeon bounds or any `dFlags`.

Comment (AJenbo) [reply Source/missiles.cpp:2380]: Making functions public is allow, yeah I didn't really remember the name but there is definitely a function like this that gets used during navigation

**Pull request merged**